public/app/plugins/datasource/influxdb/components/RawInfluxQLEditor.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FC, useState, useEffect } from 'react';
+import { TextArea, InlineFormLabel, Input, Select, HorizontalGroup } from '@grafana/ui';
+import { SelectableValue } from '@grafana/data';
+import { ResultFormat, InfluxQuery } from '../types';
+
+const RESULT_FORMATS: Array<SelectableValue<ResultFormat>> = [
+  { label: 'Time series', value: 'time_series' },
+  { label: 'Table', value: 'table' },
+  { label: 'Logs', value: 'logs' },
+];
+
+const DEFAULT_RESULT_FORMAT: ResultFormat = 'time_series';
+
+type Props = {
+  query: InfluxQuery;
+  onQueryChange: (query: InfluxQuery) => void;
+  onRunQuery: () => void;
+};
+
+// we handle 3 fields: "query", "alias", "resultFormat"
+// "resultFormat" changes are applied immediately
+// "query" and "alias" changes only happen on onblur
+export const RawInfluxQLEditor: FC<Props> = ({ query, onQueryChange, onRunQuery }) => {
+  const [currentQuery, setCurrentQuery] = useState(query.query);
+  const [currentAlias, setCurrentAlias] = useState(query.alias);
+
+  useEffect(() => {
+    // if the value changes from the outside, we accept it
+    if (currentQuery !== query.query) {
+      setCurrentQuery(query.query);
+    }
+  }, [query.query]);
+
+  useEffect(() => {
+    // if the value changes from the outside, we accept it
+    if (currentAlias !== query.alias) {
+      setCurrentAlias(query.alias);
+    }
+  }, [query.alias]);
+
+  const applyDelayedChangesAndRunQuery = () => {
+    onQueryChange({
+      ...query,
+      query: currentQuery,
+      alias: currentAlias,
+    });
+    onRunQuery();
+  };
+
+  return (
+    <div>
+      <TextArea
+        rows={3}
+        spellCheck={false}
+        placeholder="InfluxDB Query"
+        onBlur={applyDelayedChangesAndRunQuery}
+        onChange={(e) => {
+          setCurrentQuery(e.currentTarget.value);
+        }}
```

Comment:
Nit: try to avoid anonymous arrow functions as React recreates them on every render.

Reply:
thanks, yes, you are right this will get recreated at every render.... though for now i'd prefer to leave it like this, because it makes the code simpler (the alternative would be to make it use `useCallback` and that brings in some complexity), of course, if this becomes a slowdown, i'll switch it to `useCallback`.